subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/BNDSmokeTest.groovy

```diff
@@ -0,0 +1,481 @@
+    implementation "org.apache.commons:commons-lang3:$commonsVersion"
+}
+
+tasks.named("jar") {
+    bundle {
+        properties.empty() // Make this work with CC per: https://github.com/bndtools/bnd/blob/master/gradle-plugins/README.md#gradle-configuration-cache-support
+
+        bnd('Import-Package': 'org.apache.*')
+    }
+}
+"""
+
+        file("src/main/java/com/example/Example.java") << """
+package com.example;
+
+import org.apache.commons.lang3.StringUtils;
+
+public class Example {
+    public boolean testIsEmpty(String someString) {
+        return StringUtils.isEmpty(someString);
+    }
+}
+"""
+
+        when:
+        runner("jar")
+            .forwardOutput()
+            .expectLegacyDeprecationWarningIf(
+                VersionNumber.parse(GradleVersion.current().version) > VersionNumber.parse("8.1"),
+                "The AbstractTask.getConvention() method has been deprecated. " +
+                "This is scheduled to be removed in Gradle 9.0. " +
+                "Consult the upgrading guide for further information: " +
+                "https://docs.gradle.org/${GradleVersion.current().version}/userguide/upgrading_version_8.html#deprecated_access_to_conventions")
+            .build()
+
+        then: "version numbers exist in the manifest"
+        assertJarManifestContains("Import-Package", "org.apache.commons.lang3;version=\"$calculatedCommonsVersionRange\"")
+    }
+
+    def "BND plugin writes transitive external dependency versions of a direct project dependency to the manifest"() {
+        given:
+        def commonsVersion = "3.12.0"
+        def directVersion = "2.7-SNAPSHOT"
+        def calculatedDirectVersionRange = "[2.7,3)"
+        settingsFile << """
+include "direct"
+"""
+        buildFile << """
+plugins {
+    id "biz.aQute.bnd.builder"
+}
+
+${mavenCentralRepository()}
+
+dependencies {
+    implementation project(":direct")
+}
+
+tasks.named("jar") {
+    bundle {
+        properties.empty() // Make this work with CC per: https://github.com/bndtools/bnd/blob/master/gradle-plugins/README.md#gradle-configuration-cache-support
+
+        bnd('Import-Package': 'com.example.util.*')
+    }
+}
+"""
+
+        file("src/main/java/com/example/Example.java") << """
+package com.example;
+
+import com.example.util.MyUtil;
+
+public class Example {
+    public boolean testIsEmpty(String someString) {
+        return MyUtil.myIsEmpty(someString);
+    }
+}
+"""
+
+        and:
+        file("direct/build.gradle") << """
+plugins {
+    id "java-library"
+    id "biz.aQute.bnd.builder"
+}
+
+group = "com.example.direct"
+version = "$directVersion"
+
+${mavenCentralRepository()}
+
+dependencies {
+    api "org.apache.commons:commons-lang3:$commonsVersion"
+}
+
+jar {
+    bundle {
+        properties.empty() // Make this work with CC per: https://github.com/bndtools/bnd/blob/master/gradle-plugins/README.md#gradle-configuration-cache-support
+
+        bnd(\"\"\"-exportcontents: com.example.util.*\"\"\")
```

Comment:
Does this need triple quotes to work?  Would triple singlequotes work without escaping?

Reply:
I'll switch these to use the map syntax, which is a little more clear.